### ISSUE TITLE
Why don't use PhoneNumberUtil for getFullNumber?

### ISSUE THREAD

**Issue by rolgalan** [hbb20/CountryCodePickerProject#305]
HI there!

First I want to thank you for this amazing library, it works really fine!

I've recently found an issue with Ivory Coast phone numbers using your library.
Somehow, the E.164 formatting for this country **should have the leading zero**. At least it's what the base Google phone library is expecting.

In this code I've chosen Ivory Coast (CI) and Switzerland (CH) as both countries have national numbers with leading zeroes. [I'm using exactly the same number for Switzerland that was used in a previous issue you had](https://github.com/hbb20/CountryCodePickerProject/issues/183).

I'm also using the same [MichaelRocks/libphonenumber-android](https://github.com/MichaelRocks/libphonenumber-android) port of the Android libphonenumber you use as part of your library.

```kotlin
        val phoneUtil = PhoneNumberUtil.createInstance(context)

        val numberCIToParse = "07889966"
        val phoneCI = phoneUtil.parse(numberCIToParse, "CI")
        val formattedCI = phoneUtil.format(phoneCI, PhoneNumberUtil.PhoneNumberFormat.E164)

        val numberCHToParse = "0791234567"
        val phoneCH = phoneUtil.parse(numberCHToParse, "CH")
        val formattedCH = phoneUtil.format(phoneCH, PhoneNumberUtil.PhoneNumberFormat.E164)
```

If you run this piece of code, CI preserves the leading 0, whereas CH is not (which is expected). The values after formatting both numbers using the library are:
- `formattedCI` has `+22507889966`
- `formattedCH` has `+41791234567` 

However, in your method [getFullNumber()](
https://github.com/hbb20/CountryCodePickerProject/blob/549bc81eaecc65e12471b99c3f699d785cd3fcf4/ccp/src/main/java/com/hbb20/CountryCodePicker.java#L1749) (that is called by `getFullNumberWithPlus()` which I was using to retrieve the user phone and send it to my backend) you're not relying on the PhoneNumberUtils library, and you are composing the number yourselves:

`fullNumber = "" + phoneNumber.getCountryCode() + phoneNumber.getNationalNumber();`

So, instead of formatting CI numbers as `+22507889966` (valid international format according to the library), you are skipping the leading zero, returning `+2257889966`

Why do not trust in their library for this task, and handling this manually on your side?

I have to say that it was a big surprise to myself. We validate our phone numbers with a 3rd party verification service and it was reporting that the number without the leading zero was wrong.

Also, your widget properly identifies the numbers both in national and international formats, but it fails when trying to validate the Ivory Coast number without the leading 0. You can see it in these two videos:

![IvoryCoastPhone](https://user-images.githubusercontent.com/14839961/56677805-966da480-66c1-11e9-91b8-b1114027f3a1.gif)

![SwitzerlandPhone](https://user-images.githubusercontent.com/14839961/56677807-966da480-66c1-11e9-9c58-d1e8b4f8b6bb.gif)

Thanks!

**Comment by hbb20** [comment]
Thanks for reporting. Truly speaking I didn't know about valid international number with leading 0 so I did not test it either. And you are also right about using PhoneNumberUtil instead of handling it manually. I have fixed the issue in version 2.2.9 so that you can fix it in your app ASAP. Let me know if it's working as expected.


Again thanks for reporting and suggesting the solution. We appreciate it.

**Comment by rolgalan** [comment]
Amazing! This was super fast! :)
I've already tried and it works great.

Thanks!
